### DISCUSSION TITLE
Problem flash Sonoff 4chR2  "fatal error'

### DISCUSSION THREAD

**Discussion by ericson89** [arendst/Tasmota#13550] [Q&A] [answered]
Hello.
Hope to be in the right place.
I want to flash my Sonoff 4ch R2.
Before I want to save the firmware but it does not work. It seems to start well but it crashes.
Here are screenshots.
I searched the internet but I couldn't find anything.
I have two cards and I have the same with both.
if you have any advice, thanks
I write with the google translator (I am French)
<img width="859" alt="2021-11-03 11_40_26-Administrateur _ Windows PowerShell" src="https://user-images.githubusercontent.com/93651789/140050477-489d07b5-507a-4e76-a356-89f9a55d8c57.png">
![IMG_20211103_112818](https://user-images.githubusercontent.com/93651789/140050482-0786433c-9244-4736-9322-2de7e90608e5.jpg)
![IMG_20211103_112823](https://user-images.githubusercontent.com/93651789/140050535-983820f4-05d4-4952-8ff0-0458db6bd294.jpg)
![IMG_20211103_112829](https://user-images.githubusercontent.com/93651789/140050616-3ef0d052-6a16-4c98-9fc1-2a9ccc039563.jpg)

**Comment by Jason2866** [comment, marked as answer]
It looks like that your USB-serial adapter does not deliver enough "juice" (current) for a successfull flash.
More infos and recommendations https://tasmota.github.io/docs/Getting-Started/#serial-to-usb-adapter

**Reply by ericson89** [reply]
ok thanks for the help. I will try to find an external 3.3v power supply or change FTDI
Thank you

**Comment by ericson89** [comment]
I just tested by taking the 3.3v on an Arduino MEGA 2560 which is connected to USB in 5v ...
But it does exactly the same.

**Comment by ericson89** [comment]
I tried with a NODEMCU and ......
IT WORKS !!!
Great ...
Thank you very much for the link.
